File: backend/app/routers/coverage.py
"""
Coverage dashboard endpoint (admin-only).

GET /api/admin/coverage — one aggregated per-destination coverage payload:
immigration facts (approved / pending / rejected) plus provider capabilities per
service category (live / total). Cached in-process; pass ``?refresh=true`` to
force a live recompute against the current database.
"""
from typing import Any, Dict

from fastapi import APIRouter, Depends

from ..auth_deps import require_admin
from ..services.coverage_service import get_coverage_summary

router = APIRouter(prefix="/api/admin", tags=["admin-coverage"])


@router.get("/coverage")
def get_coverage(
    refresh: bool = False,
    _admin: Dict[str, Any] = Depends(require_admin),
) -> Dict[str, Any]:
    return get_coverage_summary(refresh=refresh)

File: frontend/src/api/coverage.ts
import { apiGet } from './client';

/**
 * Admin coverage dashboard (admin-only). One aggregated per-destination payload:
 * immigration facts split by review_status, and provider capabilities per service
 * category split by vetting status. Served from an in-process cache on the backend;
 * `getCoverage(true)` hits `?refresh=true` to force a live recompute.
 */
export interface CoverageCountry {
  iso: string;
  name: string;
  flag: string | null;
  catalog_name: string | null;
  facts: { approved: number; pending: number; rejected: number; total: number };
  providers: {
    by_cat: Record<string, number>;
    approved: number;
    pending: number;
    total: number;
  };
}

export interface CoverageTotals {
  destinations: number;
  facts_approved: number;
  facts_pending: number;
  facts_rejected: number;
  facts_total: number;
  caps_total: number;
  caps_approved: number;
  caps_pending: number;
  suppliers: number;
  expert_verified: number;
}

export interface CoverageSummary {
  generated_at: string;
  serving_categories: string[];
  countries: CoverageCountry[];
  totals: CoverageTotals;
}

export async function getCoverage(refresh = false): Promise<CoverageSummary> {
  return apiGet<CoverageSummary>(`/api/admin/coverage${refresh ? '?refresh=true' : ''}`);
}
